x/subspaces/types/permissions.go

```diff
@@ -1,139 +1,149 @@
-
-	// PermissionDeleteSubspace allows users to delete the subspace.
-	PermissionDeleteSubspace = Permission(0b100000)
-
-	// PermissionManageSections allows users to manage subspace sections.
-	PermissionManageSections = Permission(0b1000000)
-
-	// PermissionInteractWithContent allows users to interact with content inside the subspace (eg. polls)
-	PermissionInteractWithContent = Permission(0b10000000)
-
-	// PermissionEditOwnContent allows users to edit their own content inside the subspace
-	PermissionEditOwnContent = Permission(0b100000000)
-
-	// PermissionReportContent allows users to report contents
-	PermissionReportContent = Permission(0b1000000000)
-
-	// PermissionDeleteOwnReports allows users to delete existing reports made by their own
-	PermissionDeleteOwnReports = Permission(0b10000000000)
-
-	// PermissionManageReports allows users to manage other users reports
-	PermissionManageReports = Permission(0b100000000000)
-
-	// PermissionManageReasons allows users to manage a subspace reasons for reporting
-	PermissionManageReasons = Permission(0b1000000000000)
+	PermissionSetPermissions = RegisterPermission("set permissions")
 
 	// PermissionEverything allows to do everything.
 	// This should usually be reserved only to the owner (which has it by default)
-	PermissionEverything = Permission(0b1111111111111)
+	PermissionEverything = RegisterPermission("everything")
 )
 
 var (
-	permissionsMap = map[Permission]string{
-		PermissionNothing:             "Nothing",
-		PermissionWrite:               "Write",
-		PermissionModerateContent:     "ModerateContent",
-		PermissionChangeInfo:          "ChangeInfo",
-		PermissionManageGroups:        "ManageGroups",
-		PermissionSetPermissions:      "SetPermissions",
-		PermissionDeleteSubspace:      "DeleteSubspace",
-		PermissionInteractWithContent: "InteractWithContent",
-		PermissionEditOwnContent:      "EditOwnContent",
-		PermissionReportContent:       "ReportContent",
-		PermissionDeleteOwnReports:    "DeleteOwnReports",
-		PermissionManageReports:       "ManageReports",
-		PermissionManageReasons:       "ManageReasons",
-		PermissionEverything:          "Everything",
-	}
+	// registeredPermissions represents the list of permissions that are registered and should be considered valid
+	registeredPermissions []Permission
 )
 
-// ParsePermission parses the given permission string as a single Permissions instance
-func ParsePermission(permission string) (Permission, error) {
-	// Check inside the map if we have anything here
-	for permValue, permString := range permissionsMap {
-		if strings.EqualFold(permission, permString) {
-			return permValue, nil
+// Permission represents a permission that can be set to a user or user group
+type Permission = string
+
+// newPermission returns a new Permission containing the given value
+func newPermission(permissionName string) Permission {
+	return strings.ToUpper(strings.ReplaceAll(permissionName, " ", "_"))
```

Comment:
I think it is better to merge all space first to avoid the typo case like registering `write  content` then getting `WRITE__CONTENT` permission.

Reply:
I've now improved the function to also handle such cases. Thanks for spotting this! :pray: